src/CompoundExpression/src/CompoundExpressionCalculator.cpp

```diff
@@ -0,0 +1,180 @@
+////////////////////////////////////////////////////////////////////////////
+/**
+ *  @file   CompoundExpressionCalculator.cpp
+ *  @date   Tue, June 15 2021
+ *  @brief  Definitions for calculator that allows more than 2 terms
+ */
+////////////////////////////////////////////////////////////////////////////
+
+#include <regex>
+#include <thread>
+#include <utility>
+#include <mutex>
+#include <iostream>
+#include <sstream>
+
+#include "CompoundExpressionCalculator.h"
+#include "CalculatorStrings.h"
+
+namespace calculator
+{
+    CompoundExpressionCalculator::CompoundExpressionCalculator(std::string compoundExpression)
+            :compoundExpression_(std::move(compoundExpression))
+    {
+        resultExpression_ = compoundExpression_;
+        result_ = computeResult();
```

Comment:
result_ is the member that needed initialized. computeResult() in the constructor made sense since it can be a costly operation and I would rather save the answer instead of re-computing each time getResult() is called. In addition since ints are 0 by default I can't wait for getResult() is called to do the operation. I could make a isResultComputed_ flag or something like that so I can wait for the to compute until the first time getResult() is called.